readabilipy/extractors/extract_date.py

```diff
@@ -0,0 +1,163 @@
+        ('//meta[@name="Last-Modified"]/@content', 1),
+        ('//meta[@name="dcterms.created"]/@content', 1),
+        ('//meta[@name="published"]/@content', 1),
+        ('//meta[@property="og:article:published_time"]/@content', 1),
+        ('//meta[@property="og:pubdate"]/@content', 1),
+        ('//meta[contains(concat(" ", normalize-space(@property), " "), " article:published_time ")]/@content', 1),
+        ('//meta[@itemprop="datePublished"]/@content', 2),
+        ('//time[contains(@class, "entry-date")]/@datetime', 3),
+        ('//time[@class="post-entry-time"]/@datetime', 1),
+        ('//time[@itemprop="datePublished"]/@datetime', 3),
+        ('//time[@class="post-date updated"]/@datetime', 1),
+        ('//time[@class="post__date"]/@datetime', 1),
+        ('//time/@datetime', 1),
+        ('//time[@class="post__date"]/@content', 1),
+        ('//time/text()', 1),
+        ('//div[@class="article-heading-author-name"]//time[@class="timeagofunction"]/@datetime', 1),
+        ('//div[@class="keyvals"]/@data-content_published_date', 1),
+        ('//div[@class="article-byline"]/time[@class="visually-hidden"]/text()', 1),
+        ('//div/time/@datetime', 1),
+        ('//div[@class="subarticle"]/p/text()', 1),
+        ('//div[@class="text"]/p/text()', 1),
+        ('//div[@class="publish-date"]/text()', 1),
+        ('//footer[@class="byline"]/time/@datetime', 1),
+        ('//span[@class="timestamp "]/@data-epoch-time', 1),
+        ('//span[@class="article-element__meta-item"]', 1),
+        ('//span[@class="date published time"]/@title', 1),
+        ('//span[@class="updated"]/text()', 1),
+        ('//span[@class="entry-date"]/text()', 1),
+        ('//p[@itemprop="datePublished"]/text()', 1),
+        ('//p[@class="entry-byline"]//time[@class="entry-date"]/@datetime', 1),
+        ('//article[contains(@id, "post")]//time[@class="entry-date published"]/@datetime', 1),
+        ('substring-after(//*[comment()[contains(., "By")]]/comment(), "-")', 1),
+        ('substring-after(//p[@class="text-muted"]/text(), ",")', 1)
+    ]
+
+    # Get the date
+    date_string = extract_element(html, xpaths)
+
+    # Convert the date_string to a consistent format
+    # Tuple scores reflect preference of format, more specific formats should be prioritised
+    formats = [
+        ('YYYY-MM-DD hh:mm:ss', 2),
+        ('YYYY-MM-DD', 1),
+        ('ddd MMM DD YYYY hh:mm:ss', 2),
+        ('ddd MMM D YYYY HH:mm:ss', 2),
+        ('ddd MMM DD YYYY', 1),
+        ('DD/MM/YY', 1),
+        ('h:m A MM/DD/YYYY', 2),
+        ('MM/DD/YYYY', 1),
+        ('unix_milliseconds', 2),
+        ('MMM DD YYYY', 1),
+        ('MMM D, YYYY', 1),
+        ('MMMM DD, YYYY', 1),
+        ('MMMM D, YYYY', 1),
+        ('[Published] hh:mm A [EST] MMM DD, YYYY', 2)
+    ]
+
+    # See if a date of any of these formats can be found
```

Comment:
Prefer the dict solution that we used for titles

Reply:
This is using the dict solution, in the same function. See ```date_string = extract_element(html, xpaths)``` line 127

Reply:
As discussed today, we can copy the solution from `extract_element` and use it here.